src/nvim/os/unix_defs.h

```diff
@@ -31,12 +31,6 @@
 #ifndef USR_EXRC_FILE
 # define USR_EXRC_FILE "~/.exrc"
 #endif
-#ifndef USR_VIMRC_FILE
-# define USR_VIMRC_FILE "~/.nvimrc"
```

Comment:
What about wrapping nvim in a script that sets the XDG* variables as the user sees fit? I don't see a need to add more code for that specific usecase. Besides: changing those variables will mess up applications one might want to run from within nvim that expect the XDG* variables to be the same as the system's.